DcCore/DcCore/Helper/KeychainManager.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+import Security
+
+public enum KeychainError: Error {
+    case noPassword
+    case unhandledError(message: String, status: OSStatus)
+}
+
+public class KeychainManager {
+    private typealias KcM = KeychainManager
+    private static let teamId = "8Y86453UA8"
+    private static let sharedKeychainGroup = "\(KcM.teamId).group.chat.delta.ios"
+
+    public static func getAccountSecret(accountID: Int) throws -> String {
+        do {
+            return try queryAccountSecret(id: accountID)
+        } catch KeychainError.noPassword {
+            return try addAccountSecret(id: accountID)
```

Comment:
this looks a bit dangerous to me:

if `queryAccountSecret()` fails temporary for some reasons (so eg. errSecSuccess), the existing secret is overwritten and the database will get unusable.

so, we should differ carefully between "Ok, but nothing found and "Error", maybe we should change the return value for `queryAccountSecret()` for that.

Reply:
we only call `addAccountSecret()` if there's no password set at all. For all other possible error cases, the error is passed to the calling method. So no existing password will be ever overwritten.

Reply:
i see, i've overseen that bit. thanks for explanation!